js/views/userPageVw.js

```diff
@@ -1413,6 +1451,76 @@ module.exports = baseVw.extend({
 
   },
 
+  addModeratorClick: function(e){
+    var $targ = $(e.target).closest('.js-addmoderator');
+
+    $targ.addClass('loading');
+
+    var user = this.model.get('user');
```

Comment:
Since user is a reference to the page model's user object, when you add guids to user.moderators, the page model now has an invalid user.moderator object, and any other logic that depends on it will break. 

You either need to do a new fetch for the this.options.userModel to get fresh data after the save, and clone it to model.user, or construct the whole moderator object and add it to moderators.

The very kludgey way would be to refresh the page, but that's not recommended.